base/acme/src/main/java/org/dogtagpki/acme/database/ACMEDatabase.java

```diff
@@ -37,6 +37,14 @@ public void init() throws Exception {
     public void close() throws Exception {
     }
 
+    public Boolean getEnabled() throws Exception {
+        throw new NotImplementedException("ACMEDatabase.getEnabled() must be overridden by specific implementation");
+    }
+
+    public void setEnabled(Boolean enabled) throws Exception {
```

Comment:
Why not just make these abstract?

Reply:
`ACMEDatabase` has some other subclasses. If we add new abstract methods, all subclasses will have to be fixed at the same time. With default methods like above we can focus on fixing `LDAPDatabase` first, and other subclasses can be fixed separately later as needed.

Reply:
OK, that's fine.  Thanks for clarifying @edewata.

Reply:
Actually I changed this to ignore the operation instead of throwing an exception so other database types are not affected.